test/externalTests/gp2.sh

```diff
@@ -76,6 +76,9 @@ function gp2_test
     # See https://github.com/cowprotocol/contracts/issues/32
     yarn add @tenderly/hardhat-tenderly@1.1.6
 
+    # Add missing sinon type definitions
+    yarn add @types/sinon
```

Comment:
I needed explicitly install the types for the `sinon` package, since the change for some reason started to throw the following error:
```
Compiled 58 Solidity files successfully
$ tsc && tsc -p tsconfig.lib.esm.json && tsc -p tsconfig.lib.commonjs.json
test/tasks/dump.test.ts:14:40 - error TS7016: Could not find a declaration file for module 'sinon'. '/tmp/ext-test-Gnosis-Protocol-V2-ye8M6B/ext/node_modules/sinon/lib/sinon.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/sinon` if it exists or add a new declaration (.d.ts) file containing `declare module 'sinon';`

14 import { match, mock, SinonMock } from "sinon";
                                          ~~~~~~~

test/tasks/withdraw.test.ts:5:33 - error TS7016: Could not find a declaration file for module 'sinon'. '/tmp/ext-test-Gnosis-Protocol-V2-ye8M6B/ext/node_modules/sinon/lib/sinon.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/sinon` if it exists or add a new declaration (.d.ts) file containing `declare module 'sinon';`

5 import { mock, SinonMock } from "sinon";
                                  ~~~~~~~

test/tasks/withdrawService.test.ts:6:40 - error TS7016: Could not find a declaration file for module 'sinon'. '/tmp/ext-test-Gnosis-Protocol-V2-ye8M6B/ext/node_modules/sinon/lib/sinon.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/sinon` if it exists or add a new declaration (.d.ts) file containing `declare module 'sinon';`

6 import { mock, SinonMock, match } from "sinon";
                                         ~~~~~~~
```

Not really sure why. Need to investigate how it is related a bit more.

Reply:
Maybe related with https://github.com/DefinitelyTyped/DefinitelyTyped/issues/63626?